solr/solr-ref-guide/src/nested-documents.adoc

```diff
@@ -0,0 +1,321 @@
+   In this mode relationship types(field names) between parents and their children are not saved. +
+   In this case <<nested-documents.adoc#child-doc-transformer,[child]>> transformer returns all children under the `\_childDocuments_` field.
+ * Typically you should have a field that differentiates a root doc from any nested children. However this isn't strictly necessary; so long as it's possible to write a query that can select only root documents somehow. Such a query is needed for the <<other-parsers.adoc#block-join-query-parsers,block join query parsers>> and <<nested-documents.adoc#child-doc-transformer,[child]>> doc transformer to function.
+
+=== XML Examples
+
+For example, here are two documents and their child documents.
+It illustrates two styles of adding child documents; the first is associated via a field "comment" (preferred),
+and the second is done in the classic way now referred to as an "anonymous" or "unlabelled" child document.
+This field label relationship is available to the URP chain in Solr but is ultimately discarded.
+Solr 8 will save the relationship.
+
+[source,xml]
+----
+<add>
+  <doc>
+    <field name="id">1</field>
+    <field name="title">Solr adds block join support</field>
+    <field name="content_type">parentDocument</field>
+    <field name="content">
+      <doc>
+        <field name="id">2</field>
+        <field name="comments">SolrCloud supports it too!</field>
+      </doc>
+    </field>
+  </doc>
+  <doc>
+    <field name="id">3</field>
+    <field name="title">New Lucene and Solr release is out</field>
+    <field name="content_type">parentDocument</field>
+    <doc>
+      <field name="id">4</field>
+      <field name="comments">Lots of new features</field>
+    </doc>
+  </doc>
+</add>
+----
+
+In this example, we have indexed the parent documents with the field `content_type`, which has the value "parentDocument".
+We could have also used a boolean field, such as `isParent`, with a value of "true", or any other similar approach.
+
+=== JSON Examples
+
+This example is equivalent to the XML example above.
+Again, the field labelled relationship is preferred.
+The labelled relationship here is one child document but could have been wrapped in array brackets.
+For the anonymous relationship, note the special `\_childDocuments_` key whose contents must be an array of child documents.
+
+[source,json]
+----
+[
+  {
+    "id": "1",
+    "title": "Solr adds block join support",
+    "content_type": "parentDocument",
+    "comments": [{
+        "id": "2",
+        "content": "SolrCloud supports it too!"
+      },
+      {
+        "id": "3",
+        "content": "New filter syntax"
+      }
+    ]
+  },
+  {
+    "id": "4",
+    "title": "New Lucene and Solr release is out",
+    "content_type": "parentDocument",
+    "_childDocuments_": [
+      {
+        "id": "5",
+        "comments": "Lots of new features"
+      }
+    ]
+  }
+]
+----
+
+.Root Only Mode
+[NOTE]
+====
```

Comment:
I really don't know why, but for some reason the `====` here are causing the content inside the block to come out as monospaced, which I don't think you want. Since the content for the Note does not contain any empty lines, you can omit them at the start and end and the content comes out as normal text. Again, I really don't know why - the `====` is supposed to be the syntax to use for a NOTE block, but somehow here it's going awry.